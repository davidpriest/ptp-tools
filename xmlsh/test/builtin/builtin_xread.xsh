# builtin xread command
# reads input into a single xml variable
# input is expected in XML format (not xquery format, no sequences)

# read from a hear document
xread a <<EOF
<test>data</test>
EOF

echo a is $a
# test type with xquery on set
echo type of a is $(set | xquery '//variable[@name="a"]/@type/string()')


# read from a file 
cd ../../samples/data

xread b < books.xml
# Test type by test -X 
[ -X $b ] && echo document is XML type
exit 0

