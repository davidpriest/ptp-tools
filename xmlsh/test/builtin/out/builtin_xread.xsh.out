a is <test>data</test>
type of a is xml
document is XML type
